brave/brave-ios
# Ref #4879: VPN Churn Improvements - Callout Logic and Settings Changes

## Summary of Changes

This is 3rd and last part of churn improvements for receipt retry state. It is using the verifyReceipt Data method implemented in https://github.com/brave/brave-ios/pull/7803 and also adding logic to show update payment callout for update payment status the callouts are added with https://github.com/brave/brave-ios/pull/7670


This pull request fixes #4879

## Submitter Checklist:


## Test Plan:

PART 1: Active subscription

- Purchase VPN for 1 month on sandbox account - check it is working properly
- Kill the Browser and ran again 
- In details you should be able to see VPN subscription is active

PART 2: Stop auto-renewable

- Go Appstore -> Subscriptions -> Sandbox Account -> Manage
- Toggle off Allow Purchases & Renewals
- Press done
- This will stop renewing subscription and will put app in grace period

PART 3: Test if Retry logic with grave period is performing (which it doesnt)

- Run the app again
- This time app should be in grace period of some sort after subscription expires (~5min) and you should ne able to see grace period elements in UI
-Custom Alert in the beginning will be shown for devices under iOS 16

## Screenshots:

![IMG_3037](https://github.com/brave/brave-ios/assets/6643505/1d5358fb-79e8-4e03-90a5-8cbfcdc3c375)

https://github.com/brave/brave-ios/assets/6643505/6cda1f6e-2124-413d-91f1-1774452a6f27


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
